### PR TITLE
Validate Twilio requests to ensure security

### LINKED ISSUE #17: Validate Twilio responses
https://www.twilio.com/docs/guides/how-to-secure-your-express-app-by-validating-incoming-twilio-requests

### PR DESCRIPTION
**Pull request by jschairb**
I added the twilio-node middleware that will automatically check request
signatures that come from Twilio. It will only check the signature for the
production environment.

Closes #17

### CODE REVIEW

File: config/index.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ module.exports = {
     // Twilio Auth Token - found on your dashboard
     authToken: process.env.TWILIO_AUTH_TOKEN,
 
+    // Simple accessor for Environment
+    env: process.env.NODE_ENV,
```
Comment (flazz) [inline config/index.js]: just did some reading on this, i'm critical of this strategy but this is just a POC and something express specific right?

File: routes/index.js [hunk 2]
```diff
@@ -11,6 +11,10 @@ var VoiceResponse = twilio.twiml.VoiceResponse;
 // Create a Twilio REST API client for authenticated requests to Twilio
 var twilio_client = twilio(config.accountSid, config.authToken);
 
+// Validate Twilio Requests in production using Express middleware
+// See: https://www.twilio.com/docs/api/security#validating-requests
+const shouldValidate = config.env == 'production';
```
Comment (flazz) [inline routes/index.js]: is this better put into config.js?

File: routes/index.js [hunk 3]
```diff
@@ -75,7 +79,7 @@ module.exports = function(app) {
     });
 
     // This must come before /callbacks/:uuid to be matched explictly.
-    app.post('/callbacks/ctc-agent-answers', (request, response) => {
+    app.post('/callbacks/ctc-agent-answers', twilio.webhook({validate: shouldValidate}), (request, response) => {
```
Comment (flazz) [inline routes/index.js]: is this like middleware?

**Review by flazz — APPROVED** [review]
a few minor comments, no deal breakers

File: routes/index.js (hunk 2 above)
Comment (jschairb) [reply routes/index.js]: Yes, I think that's appropriate.

File: config/index.js (hunk 1 above)
Comment (jschairb) [reply config/index.js]: In what way do you see a risk? Is it relying on an environment being set or putting it in the config?

File: routes/index.js (hunk 3 above)
Comment (jschairb) [reply routes/index.js]: Yes, it's middleware provided by [twilio-node](https://github.com/twilio/twilio-node). You can read about it in [lib/webhooks/webhooks.js](https://github.com/twilio/twilio-node/blob/master/lib/webhooks/webhooks.js#L92).

**Pull request merged**